### PR TITLE
Clean up and fix the README-EDITING.md

### PR DESCRIPTION
**Pull request by Siemienik**
Clean up and fix the README-EDITING.md  
Describe in-version linking

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/8420
2.
3.

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

[skip changelog]

### CODE REVIEW

File: docs/README-EDITING.md [hunk 1]
```diff
@@ -125,6 +103,19 @@ To remove an existing version of the Handsontable documentation:
...
+Linking page in a single version requires using `@`-syntaxt. For instance: `[Core](@/api/core.md)`.
+* After the `@` should be provided relative file path from the current version root dir.
+* **Target file should define a permalink in Frontmatter section.**
+* File path should contain an extension (`.md`).
+* It is allowed to use anchors: `[Core](@/api/core.md#some-anchor)`.
+* If generating final URL fails, then it outputs the initial value as a relative link. That is caught then by the linking-checker tool.
+
+Many examples you may find by searching `](@/` in the `next` directory.
+
+Please try to avoid linking using absolute or relative URLs. It is much better to use the `@`-syntaxt
```
Comment (kirszenbaum) [inline docs/README-EDITING.md:117]:
```suggestion
## Markdown links

When linking to other documentation pages, avoid using absolute links or relative URLs.

To link to another page in the same documentation version, use the following syntax:

```markdown
[link_text](@/relative_file_path_from_this_version's_root/file_name.md#some-anchor)
\```

For example, to link to a file called `core.md`, from anywhere in the same documentation version:

```markdown
[Core](@/api/core.md)
\```

Follow these rules:
* After the `@` character, provide the target's relative file path (from the current version's root directory).<br>
  For example: `[Clipboard][@/guides/cell-features/clipboard.md]`.
* After the target file's name, add the `.md` [extension](#filenames)<br>
  For example: `[Autofill](@/api/autofill.md)`.
* To link to a specific section, use anchors.<br>
  For example: `[Core](@/api/core.md#some-anchor)`.

Also, the following rules apply:
* The target file needs to have the `permalink` [frontmatter](#frontmatter) tag defined.
* If generating a final URL link fails, the initial value gets output as a relative link.<br>
  The documentation's [link checker](./README.md#documentation-npm-scripts) catches such failed links.
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]
@Siemienik - in the suggestion, I needed to escape markdown in two places (with `\`).

File: docs/README-EDITING.md (hunk 1 above)
Comment (Siemienik) [reply docs/README-EDITING.md]: Great suggestion, thank you  🥇

**Review by kirszenbaum — APPROVED** [review]

**Review by aninde — APPROVED** [review]
Thanks awfully! 👍

**Pull request merged**